### ISSUE TITLE
consumer: re-use http client or disable keep alives

### ISSUE THREAD

**Issue by zalecn** [nsqio/go-nsq#169]
apiRequestNegotiateV1:

```
httpclient := &http.Client{Transport: newDeadlineTransport(2 * time.Second)}
```

newDeadlineTransport:

```
func newDeadlineTransport(timeout time.Duration) *http.Transport {
    transport := &http.Transport{
        Dial: func(netw, addr string) (net.Conn, error) {
            c, err := net.DialTimeout(netw, addr, timeout)
            if err != nil {
                return nil, err
            }
            return &deadlinedConn{timeout, c}, nil
        },
    }
    return transport
}
```

net/http doesn't immediately close the connection if the transport DisableKeepAlives no set true.
if you want to reuse the connection, must keep the same transport object, because golang put idle connection in the transport object.

**Comment by mreiferson** [comment]
@zhumingjian1229 interesting, thanks for the report!

**Comment by mreiferson** [comment]
fixed in #209
